ecl/hql/hqlcollect.cpp

```diff
@@ -548,10 +548,16 @@ void FileSystemDirectory::processDependencies(IPropertyTree * dependTree, const
     {
         IPropertyTree & cur = depends->query();
         const char * name = cur.queryName();
+
         // The special node __empty__ is added for an emptry string mapping "" =>, which is the entry for this package
-        if (isEmptyString(name) || strieq(name, "__empty__"))
+        if (isEmptyString(name) || strieq(name, "__empty__") || strieq(name, "_0"))
             continue;
 
+        //npm generates entries in the format "node_modules/package", so skip the leading "node_modules/"
+        const char * encodedNodeModules = "node__modules_f"; // "node_modules/"
```

Comment:
I'm confused by this.  The comment  refers to skipping leading "node_modules/" but the code is skipping the leading "node__modules_f".  Please can you clarify what's happening here?

Reply:
Tony's PR encodes expressions that could not previously be encoded as xml.  So a tag cannot contain / so that is expanded to _f.  Because the tag is encoded _ is also expanded to __.  So encodedNodeModules is the prefix it will have been encoded to.  There may be a potential bug if the module name contains an underscore.  I'll check/fix.